library/firewall_lib.py

```diff
@@ -323,28 +311,33 @@ def handle_interface_permanent(
 
 pci_ids = None
 
+
 def get_interface_pci():
     pci_dict = {}
     interfaces = nm_get_interfaces()
     for interface in interfaces:
-        #udi/device/[vendor, device]
+        # udi/device/[vendor, device]
         interface_ids = []
         device_udi = nm_get_client().get_device_by_iface(interface).get_udi()
         device_path = os.path.join(device_udi, "device")
         for field in ["vendor", "device"]:
             with open(os.path.join(device_path, field)) as _file:
                 interface_ids.append(_file.readline().strip(" \n")[2:])
-
-        pci_dict[":".join(interface_ids)] = interface
+        interface_ids = ":".join(interface_ids)
+        if interface_ids not in pci_dict:
+            pci_dict[interface_ids] = [interface]
+        else:
```

Comment:
Maybe detailed commit message is needed for explaining why this commit is needed.

Reply:
I updated the PR summary, does that suffice?

Reply:
Well, here is a good article about how to write the commit message, https://cbea.ms/git-commit/, you can take a look.

Reply:
The project has a git commit message section in the contribution guide https://linux-system-roles.github.io/contribute.html
"Write a good commit message"

Reply:
@richm @liangwen12year I see there's an updated commit message, it looks good. What do you folks think?

Reply:
The updated commit message lgtm.